rclcpp/test/rclcpp/strategies/test_allocator_memory_strategy.cpp

```diff
@@ -351,6 +351,70 @@ class TestAllocatorMemoryStrategy : public ::testing::Test
     return ::testing::AssertionSuccess();
   }
 
+  ::testing::AssertionResult TestGetNextEntity(
+    std::shared_ptr<rclcpp::Node> node_with_entity1,
+    std::shared_ptr<rclcpp::Node> node_with_entity2,
+    std::function<rclcpp::AnyExecutable(WeakNodeList)> get_next_entity_func)
+  {
+    WeakNodeList nodes;
+    auto basic_node = std::make_shared<rclcpp::Node>("basic_node", "ns");
+    nodes.push_back(node_with_entity1->get_node_base_interface());
+    nodes.push_back(basic_node->get_node_base_interface());
+    allocator_memory_strategy()->collect_entities(nodes);
+
+    rclcpp::AnyExecutable result = get_next_entity_func(nodes);
+    if (result.node_base != node_with_entity1->get_node_base_interface()) {
+      return ::testing::AssertionFailure() <<
+             "Failed to get expected entity with specified get_next_*() function";
+    }
```

Comment:
Hello @brawner , is there a particular reason this test checks for insertion ordering of service handles? The reason I am asking, in PR #1218 ,  a map is used to store callback groups and nodes, so the assumption of insertion order is invalid in this case.

Reply:
So this file was just updated in #1252, but this part wasn't touched. However, if the iteration order cannot be guaranteed, then `basic_node` here should be created with the new method `create_node_with_disabled_callback_groups`. That should cause it to be skipped if it's visited first.

Reply:
Yes that worked. Thanks @brawner . I wanted to make sure that by removing the assumption of iteration order, I was not killing the test.